framework/behaviors/AttributeTypecastBehavior.php

```diff
@@ -233,6 +233,10 @@ public function typecastAttributes($attributeNames = null)
             }
         }
 
+        $attributeTypes = array_intersect_key(
+            $attributeTypes,
+            array_flip($this->owner->activeAttributes())
+        );
```

Comment:
That changes existing behavior. Not sure if it may lead to breaks in existing apps.

Reply:
@samdark 
> That changes existing behavior.

How fix logic errors in this case? Current version change not active attributes, but they can be undefined or have another types.

Reply:
At least that should be mentioned in `UPGRADE.md`.

Reply:
This is not a solution for undefined attributes, as this is clearly a BC break and in many scenarios this new behavior does not make sense.

> How fix logic errors in this case? Current/old version changes inactive attributes - they can be undefined, have another types. For example, model have boolean attribute defined only in some scenarios and method can be call at any scenario and checks "is this attribute false?"

Can you add a test case for this (with example model)? I'm not sure in what scenario this new behavior would be justified. If you have types defined for your attributes, why would you use `AttributeTypecastBehavior` in the first place?

Reply:
If you think there might be possible BC breaks, don't change it.

Kind of related: https://github.com/yiisoft/yii2/issues/19546

Reply:
@rob006 @schmunk42 added property to customize this behavior